apps/dashboard/app/apps/router.rb

```diff
@@ -0,0 +1,33 @@
+class Router
+
+  # Return a Router [SysRouter, UsrRouter or DevRouter] based off
+  # of the input token. Returns nil if nothing is parsed correctly.
+  #
+  # return [SysRouter, UsrRouter or DevRouter]
+  def self.router_from_token(token)
+    type, *app = token.split("/")
+    case type
+    when "dev"
+      name, = app
+      DevRouter.new(name)
+    when "usr"
+      owner, name, = app
+      UsrRouter.new(name, owner)
+    when "sys"
+      name, = app
+      SysRouter.new(name)
+    end
+  end
+
+  def self.pinned_apps
+    # @pinned_apps ||= begin
+      Configuration.pinned_apps.to_a.map do |token|
+        router = router_from_token(token.to_s)
+        next if router.nil?
+
+        FeaturedApp.new(router)
+      end.reject(&:invalid_batch_connect_app?)
+    # end
```

Comment:
This used to be memoized but was commented out. Shall we memoize again, since we call this in 2 places?

Reply:
Got it. Just needed to figure out a way to test it. I think `Router.instance_variable_set('@pinned_apps', nil)` in the setup works.